src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -293,16 +415,16 @@ public ulong Complete()
         /// </summary>
         public void Dispose() => Close();
```

Comment:
Shouldn't we also have CloseAsync?

Also, the type implements IDisposable (via ICancelable), and should therefore also implement the new IAsyncDisposable. Let's have it implement the two disposable interfaces directly (rather than through ICancelable).

We should do the same for the COPY types.

Reply:
Seems like we still need ICancelable for this method but I can also implement `IAsyncDisposable`
```csharp        
/// <summary>
/// Begins a binary COPY FROM STDIN operation, a high-performance data import mechanism to a PostgreSQL table.
/// </summary>
/// <param name="copyFromCommand">A COPY FROM STDIN SQL command</param>
/// <returns>A <see cref="NpgsqlBinaryImporter"/> which can be used to write rows and columns</returns>
/// <remarks>
/// See http://www.postgresql.org/docs/current/static/sql-copy.html.
/// </remarks>
public NpgsqlBinaryImporter BeginBinaryImport(string copyFromCommand)
{
      if (copyFromCommand == null)
            throw new ArgumentNullException(nameof(copyFromCommand));
       if (!copyFromCommand.TrimStart().ToUpper().StartsWith("COPY"))
             throw new ArgumentException("Must contain a COPY FROM STDIN command!", nameof(copyFromCommand));

      var connector = CheckReadyAndGetConnector();
      Log.Debug("Starting binary import", connector.Id);
      connector.StartUserAction(ConnectorState.Copy);
       try
       {
            var importer = new NpgsqlBinaryImporter(connector, copyFromCommand);
            connector.CurrentCopyOperation = importer;
             return importer;
        }
        catch
        {
             connector.EndUserAction();
              throw;
        }
}

Reply:
I am not entirely sure how we will proceed here my last commit maybe headed in the right direction but I'm not 100% positive how to proceed

Reply:
Just thought to ask to implement this interfaces (: